Make "avsc" optional

abicky opened the pull request:
All Avro files have their schema information,
so "avsc" should be optional.

abicky committed:
Make "avsc" optional

All Avro files have their schema information,
so "avsc" should be optional.

joker1007 commented on src/main/java/org/embulk/parser/avro/AvroParserPlugin.java:
```diff
@@ -128,26 +133,14 @@ public void run(TaskSource taskSource, Schema schema,
-            ImmutableMap<String, BaseColumnGetter> columnGetters = columnGettersBuilder.build();
-            DatumReader<GenericRecord> reader = new GenericDatumReader<>(avroSchema);
+            ColumnGetterFactory factory = new ColumnGetterFactory(pageBuilder, timestampParsers);
+            DatumReader<GenericRecord> reader = new GenericDatumReader<>();
```
A reader can read avro file by different schema from one of itself.
For example, avro file doesn't have `new_column`, but reader can read `new_column` with default value by new schema.
Unless embulk plugin passes avroSchema object to `GenericDatumReader`, plugin user cannot use this feature.
And so we need more hacks.

abicky replied:
I see. I've supported such cases in https://github.com/joker1007/embulk-parser-avro/pull/2/commits/d792c3f7c8463c2188bea2ec65d2a0fda9d37742. How about it?

abicky committed:
Use avsc to assign default values to new fields

Avro files include their schema information,
but sometimes we want to another schema to
assign default values to new fields.

sekikn commented:
Any updates on this PR? This proposal sounds quite reasonable to me.
